lib/git.js

```diff
@@ -51,4 +51,20 @@ module.exports.remote = async function remote (opts = {}) {
   } catch (e) {
     // ignore error
   }
+}
+module.exports.remote = gitRemote;
+
+module.exports.repository = async function (cwd, pkg) {
+  let repo;
+  if (!pkg || !pkg.repository) {
+    const remote = await gitRemote({ cwd });
+    if (remote) {
+      repo = remote;
+    }
+  } else if (pkg && typeof pkg.repository === 'string') {
+    repo = pkg.repository;
+  } else if (pkg && typeof pkg.repository !== 'undefined' && pkg.repository.url) {
+    repo = pkg.repository.url;
+  }
+  return repo;
```

Comment:
if there's no remote, and no `repo`, shouldn't this reject?
```suggestion
      return remote;
    }
  } else if (pkg && typeof pkg.repository === 'string') {
    return pkg.repository;
  } else if (pkg && typeof pkg.repository !== 'undefined' && pkg.repository.url) {
    return pkg.repository.url;
  }
  // reject/throw here?
```

Reply:
That would make this breaking, but I don't see a reason to limit this to run within a repo. My thinking here is if someone runs this before creating a repo in their init flow.

Reply:
ok, then the suggestion still stands, just remove the comment :-)